core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -100,6 +103,7 @@ public Status acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
         handleNameResolutionError(unavailableStatus);
         return unavailableStatus;
       }
+      cleanServers.add(removeDuplicateAddresses(eag));
```

Comment:
If DNS returned duplicate addresses, we'd have two EAGs with one address each. This `removeDuplicateAddresses()` would do nothing to remove the duplication across those EAGs. Don't we need that?